float64 x_velocity
float64 angular_velocity